helper/schema/resource_data.go

```diff
@@ -171,6 +172,14 @@ func (d *ResourceData) SetPartial(k string) {
 	}
 }
 
+func (d *ResourceData) MarkNewResource() {
+	d.isNew = true
+}
+
+func (d *ResourceData) IsNewResource() bool {
+	return d.isNew
+}
```

Comment:
Is there any reason to not just have `IsNewResource()` check `Id() == ""` directly rather than keeping the boolean struct member?